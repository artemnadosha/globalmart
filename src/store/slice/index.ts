export {
  default as cartSlice,
  removeItemCart,
  addItemCart,
} from "./cartSlice/cartSlice";
export {
  default as modalSlice,
  responseModal,
  toggleIsActive,
} from "./modalSlice/modalSlice";
export {
  default as favoriteSlice,
  addItemFavorite,
  removeItemFavorite,
} from "./favoriteSlice/favoriteSlice";
export { default as productSlice } from "./productSlice/productSlice";
export { default as fetchProduct } from "./productSlice/fetchProduct";
export { default as fetchCategories } from "./categoriesSlice/fetchCategories";
export { default as categoriesSlice } from "./categoriesSlice/categoriesSlice";
