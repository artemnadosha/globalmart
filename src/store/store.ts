import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { cartSlice, favoriteSlice, modalSlice } from "./slice";
import { api } from "./api/api";
import { setupListeners } from "@reduxjs/toolkit/query";

const rootReducer = combineReducers({
  cart: cartSlice,
  favorite: favoriteSlice,
  modal: modalSlice,
  [api.reducerPath]: api.reducer,
});
export const store = configureStore({
  reducer: rootReducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(api.middleware),
});

setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = useDispatch<AppDispatch>;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
