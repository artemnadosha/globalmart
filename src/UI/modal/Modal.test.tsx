import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Modal } from "../index";
import { store } from "../../store/store";
import { Provider } from "react-redux";

describe("modal", () => {
  test("spapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <Modal content="test" titleHeader="test" closeButton={false} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test("modal render with props", () => {
    render(
      <Provider store={store}>
        <Modal content="content" titleHeader="title" closeButton={false} />
      </Provider>
    );

    expect(screen.getByText("content")).toBeInTheDocument();
    expect(screen.getByText("title")).toBeInTheDocument();

    const closeButton = screen.queryByTestId("close-button");
    expect(closeButton).not.toBeInTheDocument();
  });
  test("modal render with closeButton true", () => {
    render(
      <Provider store={store}>
        <Modal content="content" titleHeader="title" closeButton={true} />
      </Provider>
    );

    const closeButton = screen.queryByTestId("close-button");
    expect(closeButton).toBeInTheDocument();
  });
});
