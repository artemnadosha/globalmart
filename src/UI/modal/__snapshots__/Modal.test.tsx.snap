// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal spapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="overlay"
    />
    <div
      class="modal"
    >
      <div
        class="wrapper"
      >
        <h2>
          test
        </h2>
      </div>
      <div
        class="wrapper"
      >
        <p>
          test
        </p>
      </div>
      <div
        class="wrapper"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-htwhyh ant-btn-primary ant-btn-lg"
          style="width: 100%;"
          type="button"
        >
          <span>
            Ok
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-htwhyh ant-btn-primary ant-btn-lg ant-btn-dangerous"
          style="width: 100%;"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
